tests/core/test_pytree.py

```diff
@@ -0,0 +1,29 @@
+from functools import partial
```

Comment:
This file needs 
- license
- module docstring
- function docstrings
- to be linted
- to be added to `tests/tests_passing_pylint`

and could potentially include a test for the `__repr__` :)

Reply:
so I should have the test pass pylint?

Reply:
Yes, with the exceptions in `.pylintrc`, i.e. run `pylint --rcfile .pylintrc <file>` inside the repo folder